File: services/core/src/main/java/me/unbrdn/core/auth/application/port/in/RefreshTokenUseCase.java
package me.unbrdn.core.auth.application.port.in;

import me.unbrdn.core.auth.application.interactor.dto.command.RefreshTokenCommand;
import me.unbrdn.core.auth.application.interactor.dto.result.RefreshTokenResult;

public interface RefreshTokenUseCase {
    RefreshTokenResult execute(RefreshTokenCommand command);
}

File: services/bff/src/modules/auth/usecases/refresh-token.usecase.ts
import { Injectable } from "@nestjs/common";
import { AuthGrpcService } from "src/infra/grpc/services/auth-grpc.service";

export class RefreshTokenCommand {
    constructor(public readonly refreshToken: string) {}
}

export class RefreshTokenResult {
    constructor(
        public readonly accessToken: string,
        public readonly refreshToken: string,
    ) {}
}

@Injectable()
export class RefreshTokenUseCase {
    constructor(private readonly authGrpcService: AuthGrpcService) {}

    /* Refresh Token으로 Access Token 재발급 */
    async execute(command: RefreshTokenCommand): Promise<RefreshTokenResult> {
        const response = await this.authGrpcService.refreshToken({
            refreshToken: command.refreshToken,
        });

        return new RefreshTokenResult(response.accessToken, response.refreshToken);
    }
}
